datasets/src/zenke_2a/datagen.py

```diff
@@ -1,51 +1,51 @@
 import numpy as np
 import pandas as pd
 
-from datasets.src.zenke_2a.constants import DATA_PATH
+from datasets.src.zenke_2a.constants import TEST_DATA_PATH, TRAIN_DATA_PATH
 
 
-def generate_sequential_dataset(num_samples: int = 10000, num_clusters: int = 2, cluster_switch_prob: float = 0.01,
-                                cluster_spread: float = 0.5) -> pd.DataFrame:
+def generate_sequential_dataset(num_samples: int = 100, num_datapoints: int = 100, num_clusters: int = 2,
+                                cluster_switch_prob: float = 0.01, cluster_spread: float = 0.5) -> pd.DataFrame:
     """
-    Generates a sequential dataset with rare abrupt switches between clusters.
+    Generates a sequential dataset with rare abrupt switches between clusters, organized into batches,
+    with a defined number of samples per batch.
 
     Parameters:
-    num_samples (int): Total number of samples in the dataset.
+    num_batches (int): Number of batches.
+    samples_per_batch (int): Number of samples per batch.
     num_clusters (int): Number of clusters.
     cluster_switch_prob (float): Probability of switching to a different cluster.
     cluster_spread (float): Variability within the cluster.
 
     Returns:
-    pd.DataFrame: DataFrame containing the sequential dataset.
+    pd.DataFrame: DataFrame containing the sequential dataset with batch information.
     """
-    # Initialize variables
-    data = np.zeros((num_samples, 2))  # Two-dimensional data points
-    current_cluster = 0
+    data = []  # List to store data before converting to DataFrame
 
-    for i in range(num_samples):
-        # Decide whether to switch clusters
-        if np.random.rand() < cluster_switch_prob:
-            current_cluster = (current_cluster + 1) % num_clusters
+    for sample in range(num_samples):
```

Comment:
I wonder if making this cluster generation using a simple Gaussian Mixture model as a generative model could be advantageous. (e.g. https://probml.github.io/dynamax/notebooks/hmm/gaussian_hmm.html)

Reply:
I'd say that if we need to touch this datagen again we can change to GMM. That may allow us to not write a lot of this logic, which would be better :)